### ISSUE TITLE
const in for statement

### ISSUE THREAD

**Issue by shuhei** [babel/babel#1229]
`const` that is declared in initialization of `for` statement does not throw syntax error when it is reassigned in final expression of the same `for` statement.

```
$ babel --version
5.0.12
$ echo 'for (const i = 0; i < 3; i = i + 1) { console.log(i); }' | babel
"use strict";

for (var i = 0; i < 3; i++) {
  console.log(i);
}
```

Also `i++` does not throw.

**Comment by shuhei** [comment]
Thanks for the rapid fix! I appriciate your great work at all times.

**Comment by sebmck** [comment]
Sorry for the delay. Fixed as of 5.1.0, thanks!
